### DISCUSSION TITLE
Update detected but not applied (i.e. checked out)

### DISCUSSION THREAD

**Discussion by mat-m** [ohmyzsh/ohmyzsh#10547] [Help]
Hello,

OMZ is a subrepo of my dotfiles. I don't know if it has an impact (probably, since there is no branch in checkout, opnly a "raw" commit).

I think this line 
https://github.com/ohmyzsh/ohmyzsh/blob/116fc2feaa40bf4ce11c3ac72077a0ffba109525/tools/upgrade.sh#L202
may have an impact on actual checkout.
I guess it was set to have the changelog filled, but there is no command to go forward again.

**Comment by mcornella** [comment]
The upgrade script first checks whether you're in a branch or a detached commit. If you update when `master` is checked out, that instruction will get you back to the master branch.

If you're using submodules with your dotfiles, you should manually update OMZ using submodules and [disable the update script](https://github.com/ohmyzsh/ohmyzsh/wiki/Settings#omzupdate-mode).

**Reply by mat-m** [reply]
OK,

I will build my update script for omz & custom themes.

Thanks

**Reply by mat-m** [reply]
@mcornella zstyle commands from your link should be just passed in commandline or copied into .zshrc ?

**Reply by mcornella** [reply]
Copied into .zshrc, before ohmyzsh is sourced. This is specified at the top of the Settings section.

**Reply by mat-m** [reply]
Hello, 

I decided to keep `zstyle ':omz:update' mode reminder`, as it tells me when to lauch the update.
On last update, I ran `git submodule update --remote --recursive`, but  omz still complained I need to update, until I run

```
omz update
Updating Oh My Zsh
Fast-forwarded master to 53863e7b3ff0c2e2816e90dab3d870adebdf49c7.
master

BREAKING CHANGES:
[...]
```

Is it expected ?

**Reply by mcornella** [reply]
Is the remote properly setup? It might be that the submodule is targetting a different remote than [github.com/ohmyzsh/ohmyzsh](https://github.com/ohmyzsh/ohmyzsh).

**Reply by mat-m** [reply]
```
git submodule foreach git remote -v 
Entering 'config/omzcustom/themes/powerlevel10k'
origin	https://github.com/romkatv/powerlevel10k.git (fetch)
origin	https://github.com/romkatv/powerlevel10k.git (push)
Entering 'config/omzcustom/themes/spaceship-prompt'
origin	https://github.com/denysdovhan/spaceship-prompt.git (fetch)
origin	https://github.com/denysdovhan/spaceship-prompt.git (push)
Entering 'oh-my-zsh'
origin	https://github.com/ohmyzsh/ohmyzsh.git (fetch)
origin	https://github.com/ohmyzsh/ohmyzsh.git (push)
```

We'll see next time

**Reply by mat-m** [reply]
I was in detached HEAD in my submodule.
When you check `local_head`, you're using the `$branch` var, possibly a default "master". IMO, you should check for HEAD, which works for detached HEAD like submodules to tell there are no updates.

**Reply by mcornella** [reply]
Can you open a bug report?
